x-pack/plugins/apm/public/plugin.ts

```diff
@@ -71,21 +66,47 @@ export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
     this.initializerContext = initializerContext;
   }
   public setup(core: CoreSetup, plugins: ApmPluginSetupDeps) {
-    createCallApmApi(core.http);
     const config = this.initializerContext.config.get();
     const pluginSetupDeps = plugins;
 
     pluginSetupDeps.home.environment.update({ apmUi: true });
     pluginSetupDeps.home.featureCatalogue.register(featureCatalogueEntry);
 
     if (plugins.observability) {
+      const getApmDataHelper = async () => {
+        const { fetchOverviewPageData, hasData } = await import(
+          './services/rest/apm_overview_fetchers'
+        );
+
+        return { fetchOverviewPageData, hasData };
+      };
       plugins.observability.dashboard.register({
         appName: 'apm',
-        fetchData: fetchOverviewPageData,
-        hasData,
+        hasData: async () => {
+          const dataHelper = await getApmDataHelper();
+          return await dataHelper.hasData();
+        },
+        fetchData: async (params: FetchDataParams) => {
+          const dataHelper = await getApmDataHelper();
+          return await dataHelper.fetchOverviewPageData(params);
+        },
       });
     }
 
+    const setupLazyStuff = async (coreStart: CoreStart) => {
+      const { createCallApmApi } = await import(
+        './services/rest/createCallApmApi'
+      );
+
+      createCallApmApi(core.http);
+
+      // render APM feedback link in global help menu
+      const { setHelpExtension } = await import('./setHelpExtension');
+      const { setReadonlyBadge } = await import('./updateBadge');
```

Comment:
Is there a compelling reason why these are separate imports, and not just all in the same file?